### PR TITLE
Vmcs fields

### PR DESCRIPTION
**Pull request by iankronquist**
@darkengine-io

Here it is. I fixed the segfault by the way.

### CODE REVIEW

**Comment by s-mcf** [comment]
Was the per-core data broken? I'd prefer use of the built-in Linux way of doing it than an array indexed by core number.

**Comment by iankronquist** [comment]
I was writing a long comment justifying my decision and it just hit me that I was using the per-core data *only on the core where the init runs*. All those allocations and frees need to be moved into `rustyvisor_loader_core_load` and `rustyvisor_loader_core_unload`. I'll fix it tonight (unless you want to jump ahead and beat me to it).

**Comment by s-mcf** [comment]
Filed #36 for the per-core data stuff.

File: loader/linux.c [hunk 1]
```diff
@@ -9,8 +9,9 @@
 #define MB (0x1000 * KB)
 #define HEAP_SIZE (256 * KB)
 
+
+// FIXME: Use linux per core data macros
```
Comment (s-mcf) [inline loader/linux.c]: Is this FIXME still relevant?

**Comment by iankronquist** [comment]
Depends on #38.

**Comment by s-mcf** [comment]
Consider squashing these down before merging. The history is pretty noisy.

**Comment by iankronquist** [comment]
@darkengine-io that was a hell of a rebase, but it's mostly fine now.

**Review by s-mcf — APPROVED** [review]
Looketh good to mineself

**Pull request merged**